### PR TITLE
Speed up package build

### LINKED ISSUE #1793: Speed up conda package builds
### Current Behaviour
Conda package building is slow, especially on windows.

Options
1) Use Boa for conda package building
Boa is part of the mamba project
https://github.com/mamba-org/boa

2) Use libmamba solver with conda build
Might get most of the benefit just by switching solver
https://www.anaconda.com/blog/a-faster-conda-for-a-growing-community

3) Remove unneeded packages
We still have sarepy in the depend lists, but its only needed for build old docs, so should be in environment-dev.yaml. Check others.

### PR DESCRIPTION
**Pull request by samtygier-stfc**
### Issue
Closes #1793 

### Description

Remove sarepy from conda dependencies

Switch to boa for building

Clean ups to scripts: specify packages when creating the build-env. use conda run. 

Loose version constraint so that conda does not need to consider old versions

### Testing &  Acceptance Criteria 

Packages build on the PR are at:
https://anaconda.org/mantid/mantidimaging/files?channel=test

I have tested installing and running them on Linux and Windows

### Documentation

release notes

### CODE REVIEW

**Comment by samtygier-stfc** [comment]
On linux the last 2 package builds on main were 12:48 and 12:43. With boa 6:13.
On Windows last 2 package builds on main were  2h46:48 and 3h37:07. With boa 0h9:00

File: environment.yml
Comment (JackEAllen) [inline environment.yml:4]: RIP Dimitar, Gone but not forgotten

**Review by JackEAllen — COMMENTED** [review]
Changes look good, just need to test they install correctly before approval.

**Review by JackEAllen — APPROVED** [review]
Can confirm the speed up so happy to approve this 🥳

**Review by JackEAllen — APPROVED** [review]

**Pull request merged**
